### DISCUSSION TITLE
Question: can pypdf decrypt 128/256 AES without PyCryptodome

### DISCUSSION THREAD

**Discussion by JeffBarrette** [py-pdf/pypdf#1924] [Q&A] [answered]
The document suggests (to me) its possible. Am I misunderstanding? 
![image](https://github.com/py-pdf/pypdf/assets/10245586/9b2db076-cf40-4c2e-a161-7277b919ea3c)
Pypdf help: https://pypdf.readthedocs.io/en/stable/user/encryption-decryption.html

I can work successfully with 128-RC4 PDFs but with

128-AES:
```python
reader2 = PdfReader("AcrobatPro_128AES.pdf")
if reader2.is_encrypted:
    reader2.decrypt("Layout_128AES")
print(reader2.metadata) ###FAILS HERE
pypdf.errors.DependencyError: PyCryptodome is required for AES algorithm
```

256-AES it fails even sooner:
```python
Reader3 = PdfReader("AcrobatPro_256AES.pdf")
pypdf.errors.DependencyError: PyCryptodome is required for AES algorithm
```

**Comment by pubpub-zz** [comment, marked as answer]
In any case pypdf uses pyCryptodome for crypt/decrypt files. please install it first.

Until now pypdf was only capable of decoding AES files ; writing AES encrypted files was not possible. the PR to introduce this major improvement is in the pipe (#1918)
